src/utils/api.js

```diff
@@ -27,6 +27,10 @@ class Api {
 		}
 	}
 
+	refreshToken() {
+		return this._refreshToken().then(this.isRefreshing = false); // eslint-disable-line no-return-assign
```

Comment:
`this.isRefreshing` is set to false immediately  after `this._refreshToken` is being called - it does not wait for a promise to resolve.

Reply:
Also, why `this.isRefreshing` is not set to `false` by `_refreshToken()` ?